factomd.go

```diff
@@ -8,18 +8,237 @@ import (
 	"os"
 
 	"fmt"
-	"github.com/FactomProject/factomd/engine"
+	. "github.com/FactomProject/factomd/engine"
 	"time"
+	"runtime"
+	log "github.com/sirupsen/logrus"
+
+	"bufio"
+	"io"
+	"os/exec"
+	"strings"
+	"strconv"
+	"net"
 )
 
 func main() {
 	// uncomment StartProfiler() to run the pprof tool (for testing)
-	params := engine.ParseCmdLine(os.Args[1:])
+	params := ParseCmdLine(os.Args[1:])
+
+	if (params.StdoutLog != "" || params.StderrLog != "") {
+		handleLogfiles(params.StdoutLog, params.StderrLog)
+	}
+
+	log.Print("//////////////////////// Copyright 2017 Factom Foundation")
+	log.Print("//////////////////////// Use of this source code is governed by the MIT")
+	log.Print("//////////////////////// license that can be found in the LICENSE file.")
+
+	if !isCompilerVersionOK() {
+		log.Println("!!! !!! !!! ERROR: unsupported compiler version !!! !!! !!!")
+		time.Sleep(3 * time.Second)
+		os.Exit(1)
+	}
+
+	// launch debug console if requested
+	if (params.DebugConsole != "") {
+		launchDebugServer(params.DebugConsole)
+	}
+
+	//  Go Optimizations...
+	runtime.GOMAXPROCS(runtime.NumCPU()) // TODO: should be *2 to use hyperthreadding? -- clay
```

Comment:
GOMAXPROCS defaults to the number of cpus. Not sure what it does in a hyper threading situation, but we shouldn't need to change it.